src/odemis/gui/util/img.py

```diff
@@ -1292,6 +1340,38 @@ def chronogram_to_export_data(proj, raw):
         return spec
     else:
         # Draw spectrum bar plot
+        if vp is not None:
```

Comment:
Please factorize into a common function for both the spectrum data and time data.